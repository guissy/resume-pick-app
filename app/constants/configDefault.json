[
  {
    "name": "核心项",
    "children": [
      {
        "name": "RN",
        "alias": ["react native", "rn", "weex"],
        "score": 1.5
      },
      {
        "name": "flutter",
        "alias": ["flutter1", "flutter2", "dart"],
        "score": 1.5
      },
      {
        "name": "h5",
        "alias": [
          "webview",
          "uni-app",
          "uniapp",
          "umi-app",
          "umijs",
          "ionic",
          "apicloud",
          "mobile",
          "hybird",
          "移动端"
        ],
        "score": 1.5
      },
      {
        "name": "react",
        "alias": ["reactjs", "redux", "antd", "ant design"],
        "score": 1.5
      },
      {
        "name": "vue",
        "alias": ["vue2", "vuejs", "vue2", "element", "vant"],
        "score": 1.5
      },
      {
        "name": "vue3",
        "score": 1.5
      },
      {
        "name": "angular",
        "alias": [
          "ng",
          "ng2",
          "angularjs",
          "angular2",
          "angular4",
          "angular5",
          "angular6",
          "angular7",
          "angular8"
        ],
        "score": 1
      },
      {
        "name": "wxapp",
        "alias": ["微信", "小程序", "taro", "mpvue", "mpvue"],
        "score": 1.5
      },
      {
        "name": "node",
        "alias": ["nodejs", "express", "koa", "koa2", "egg", "next", "pm2"],
        "score": 1.5
      },
      {
        "name": "state mgr",
        "alias": [
          "redux",
          "saga",
          "mobx",
          "vuex",
          "vuetify",
          "dva",
          "事件总线",
          "eventBus",
          "event bus",
          "mvvm"
        ],
        "score": 1
      },
      {
        "name": "es6",
        "alias": [
          "es5",
          "es6",
          "es6+",
          "es2015",
          "es2016",
          "es2017",
          "es7",
          "es8",
          "es9",
          "ecmascript",
          "原生JS",
          "原生 JS"
        ],
        "score": 0.8
      },
      {
        "name": "w3c",
        "score": 0.5,
        "alias": ["mdn"]
      },
      {
        "name": "css3",
        "alias": [
          "flex",
          "grid",
          "css3",
          "rem",
          "less",
          "sass",
          "scss",
          "css-modules",
          "css in js",
          "css-in-js",
          "styled-components",
          "jss",
          "emotion",
          "tailwind",
          "纯CSS",
          "vw",
          "vh"
        ],
        "score": 0.5
      },
      {
        "name": "keyframes",
        "alias": [
          "animate.css",
          "wow.js",
          "gsap",
          "pixijs",
          "svga",
          "createjs"
        ],
        "score": 0.8
      }
    ]
  },
  {
    "name": "进阶",
    "children": [
      {
        "name": "typescript",
        "alias": ["ts"],
        "score": 1.5
      },
      {
        "name": "hooks",
        "alias": ["hook", "hoc"],
        "score": 1.5
      },
      {
        "name": "rxjs",
        "alias": ["ReactiveX", "ngrx"],
        "score": 1.5
      },
      {
        "name": "graphql",
        "alias": ["gql"],
        "score": 1.5
      },
      {
        "name": "test",
        "calc": "max",
        "alias": [
          "mocha",
          "jasmine",
          "jest",
          "cypress",
          "puppeteer",
          "unit test"
        ],
        "score": 1.5
      },
      {
        "name": "gayhub",
        "alias": ["github.com", "gitee.com"],
        "score": 1.5
      },
      {
        "name": "lint",
        "alias": [
          "tslint",
          "eslint",
          "jslint",
          "husky",
          "prettier",
          "命名规范",
          "代码规范",
          "commit规范",
          "接口规范"
        ],
        "score": 1
      },
      {
        "name": "review",
        "alias": ["pr", "git flow", "代码审查"],
        "score": 0.5
      },
      {
        "name": "blog",
        "alias": [
          "个人博客",
          "博客地址",
          "csdn.net",
          "cnblogs.com",
          "juejin.im",
          "zhihu.com",
          "jianshu.com",
          "segmentfault.com",
          "oschina.net",
          "imooc.com",
          "51cto.com",
          "gitbooks.io"
        ],
        "score": 1.5
      }
    ]
  },
  {
    "name": "其他",
    "children": [
      {
        "name": "大厂背景",
        "alias": [
          "\\b阿里巴巴\\b",
          "\\b腾讯\\b",
          "\\b百度",
          "\\b京东\\b",
          "\\b美团点评\\b",
          "\\b字节跳动\\b"
        ],
        "score": 2.5
      },
      {
        "name": "leader",
        "alias": ["组长", "负责人", "选型", "架构设计"],
        "score": 2
      },
      {
        "name": "manager",
        "alias": ["主持会议", "分配工作", "招聘"],
        "score": 2
      },
      {
        "name": "流量",
        "alias": ["([0-9一二三四五六七八九几十百千]+)\\s*(万|亿)"],
        "score": 2
      },
      {
        "name": "百分点",
        "alias": ["(减少|降低|提升|提高)(\\D{0,2})\\d+(\\.\\d+)?%"],
        "score": 2
      },
      {
        "name": "游戏引擎",
        "alias": ["白鹭", "p2", "pomelo", "cocos2d", "egret", "重力", "粒子"],
        "score": 1.5
      },
      {
        "name": "sql",
        "alias": [
          "sqlite3",
          "sqlite",
          "mysql",
          "postgre",
          "mariadb",
          "hive",
          "redis"
        ],
        "score": 1
      },
      {
        "name": "electron",
        "alias": ["nwjs"],
        "score": 1.2
      },
      {
        "name": "mock",
        "alias": ["mockjs", "rap", "模拟", "抓取"],
        "score": 0.5
      },
      {
        "name": "ejs",
        "alias": ["hygen", "swagger"],
        "score": 0.5
      },
      {
        "name": "git",
        "score": 0.5
      },
      {
        "name": "docker",
        "score": 0.5
      },
      {
        "name": "websocket",
        "alias": ["socket", "canvas", "charles"],
        "score": 0.5
      },
      {
        "name": "微服务",
        "alias": ["monorepo", "submodule", "无服务器", "severless"],
        "score": 0.5
      },
      {
        "name": "加密",
        "alias": [
          "密文",
          "支付",
          "md5",
          "sha1",
          "签名",
          "白名单",
          "jwt",
          "x-token",
          "crypto"
        ],
        "score": 0.5
      },
      {
        "name": "重构",
        "alias": [
          "防抖",
          "去抖",
          "节流",
          "虚拟化",
          "懒加载",
          "debounce",
          "throttle",
          "代码分割",
          "分包",
          "错误收集",
          "埋点",
          "preact"
        ],
        "score": 0.5
      },
      {
        "name": "Promise",
        "alias": ["async", "await", "generator"],
        "score": 0.5
      },
      {
        "name": "自定义",
        "alias": [
          "封装",
          "公共方法",
          "二次封装",
          "造轮子",
          "简化",
          "可复用",
          "自动化"
        ],
        "score": 0.5
      },
      {
        "name": "性能调优",
        "alias": ["毫秒", "攻克"],
        "score": 0.5
      },
      {
        "name": "component",
        "alias": ["基础组件", "公共组件", "通用组件", "业务组件", "组件化"],
        "score": 0.5
      },
      {
        "name": "编程范式",
        "alias": ["函数式", "haskell", "f#"],
        "score": 0.5
      },
      {
        "name": "技术博客",
        "alias": [
          "极客",
          "掘金",
          "知乎",
          "思否",
          "stackoverflow",
          "mdn",
          "w3school",
          "博客园",
          "简书",
          "CSDN"
        ],
        "score": 0.5
      },
      {
        "name": "python",
        "score": 0.5
      },
      {
        "name": "java",
        "alias": ["springcloud", "springboot", "springmvc", "mybatis"],
        "score": 0.5
      },
      {
        "name": "middleware",
        "alias": ["nacos", "eureka", "hystrix", "gateway", "zuul", "feign"],
        "score": 0.5
      },
      {
        "name": "queue",
        "alias": ["rocketmq", "rabbitmq"],
        "score": 0.5
      },
      {
        "name": "go",
        "score": 0.5
      },
      {
        "name": "rust",
        "score": 0.5
      },
      {
        "name": "cpp",
        "score": 0.5
      },
      {
        "name": "swift",
        "score": 0.5
      },
      {
        "name": "kotlin",
        "score": 0.5
      },
      {
        "name": "scala",
        "score": 0.5
      },
      {
        "name": "mongodb",
        "alias": ["sequelize", "typeorm", "orm"],
        "score": 0.5
      },
      {
        "name": "devops",
        "alias": ["shell", "nginx", "linux", "cicd"],
        "score": 0.5
      },
      {
        "name": "vim",
        "score": 0.5
      },
      {
        "name": "markdown",
        "score": 0.5
      },
      {
        "name": "ssr",
        "score": 0.5
      },
      {
        "name": "bi",
        "score": 0.5,
        "alias": ["可视化", "大屏", "echarts", "d3", "antv"]
      },
      {
        "name": "three.js",
        "score": 0.5
      },
      {
        "name": "postman",
        "score": 0.5,
        "alias": ["charles", "fiddler"]
      },
      {
        "name": "leetcode",
        "score": 0.5,
        "alias": ["力扣", "leet code"]
      },
      {
        "name": "deno",
        "score": 0.5
      },
      {
        "name": "wasm",
        "score": 0.5,
        "alias": ["web assembly"]
      },
      {
        "name": "代表作",
        "score": 0.5,
        "alias": ["项目链接", "作品链接", "(由|从)(0|零)到(1|一)"]
      },
      {
        "name": "iframe",
        "alias": ["flash", "dreamweaver", "div+css", "v-for", "v-if"],
        "score": -1.5
      },
      {
        "name": "mvc",
        "score": -0.5
      },
      {
        "name": "ajax",
        "alias": ["ajax", "jsonp", "ie6", "ie7", "ie8"],
        "score": -1
      },
      {
        "name": "jquery",
        "alias": [
          "jq",
          "zepto",
          "knockout",
          "knockoutjs",
          "extjs",
          "backbone",
          "dojo",
          "mootools",
          "emberjs",
          "yui3",
          "prototype.js"
        ],
        "score": -0.5
      },
      {
        "name": "svn",
        "alias": [
          "requirejs",
          "gulp",
          "grunt",
          "yeo",
          "yeoman",
          "seajs",
          "phonegap",
          "jasmine"
        ],
        "score": -1
      },
      {
        "name": "积极乐观",
        "alias": ["为人和蔼", "吃苦耐劳", "上进", "了解"],
        "score": -0.5
      },
      {
        "name": "熟悉",
        "score": -0.3
      }
    ]
  }
]
