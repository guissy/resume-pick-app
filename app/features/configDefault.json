[
  {
    "name": "核心项",
    "children": [
      {
        "name": "RN",
        "alias": ["react native", "rn", "weex"],
        "score": 1.5
      },
      {
        "name": "flutter",
        "alias": ["flutter", "dart"],
        "score": 1.5
      },
      {
        "name": "h5",
        "alias": [
          "webview",
          "uni-app",
          "uniapp",
          "umi-app",
          "ionic",
          "apicloud",
          "mobile",
          "hybird"
        ],
        "score": 1.5
      },
      {
        "name": "react",
        "alias": ["redux", "antd", "ant design"],
        "score": 1.5
      },
      {
        "name": "vue",
        "alias": ["vue2", "vuejs", "vue2", "element", "vant"],
        "score": 1.5
      },
      {
        "name": "vue3",
        "score": 1.5
      },
      {
        "name": "angular",
        "alias": [
          "ng",
          "ng2",
          "angular2",
          "angular4",
          "angular5",
          "angular6",
          "angular7",
          "angular8"
        ],
        "score": 1
      },
      {
        "name": "wxapp",
        "alias": ["微信", "小程序", "taro", "mpvue", "mpvue"],
        "score": 1.5
      },
      {
        "name": "node",
        "alias": ["nodejs", "express", "keo", "egg", "next", "typeorm", "pm2"],
        "score": 1.5
      },
      {
        "name": "electron",
        "alias": ["nwjs"],
        "score": 1.5
      },
      {
        "name": "state mgr",
        "alias": [
          "redux",
          "saga",
          "mobx",
          "vuex",
          "vuetify",
          "dva",
          "事件总线",
          "eventBus",
          "event bus"
        ],
        "score": 1
      },
      {
        "name": "es6",
        "alias": [
          "es5",
          "es6+",
          "es2015",
          "es2016",
          "es2017",
          "es7",
          "es8",
          "es9",
          "ecmascript"
        ],
        "score": 0.5
      },
      {
        "name": "w3c",
        "score": 0.5,
        "alias": ["mdn"]
      },
      {
        "name": "css3",
        "alias": [
          "flex",
          "grid",
          "css3",
          "rem",
          "less",
          "sass",
          "css-modules",
          "css in js",
          "css-in-js",
          "styled-components",
          "jss",
          "emotion"
        ],
        "score": 0.5
      }
    ]
  },
  {
    "name": "进阶",
    "children": [
      {
        "name": "typescript",
        "alias": ["ts"],
        "score": 1.5
      },
      {
        "name": "rxjs",
        "alias": ["ReactiveX", "ngrx"],
        "score": 1.5
      },
      {
        "name": "graphql",
        "alias": ["gql"],
        "score": 1.5
      },
      {
        "name": "test",
        "calc": "max",
        "alias": [
          "mocha",
          "jasmine",
          "jest",
          "cypress",
          "puppeteer",
          "unit test"
        ],
        "score": 1.5
      },
      {
        "name": "lint",
        "alias": ["tslint", "eslint", "jslint", "命名规范", "代码规范"],
        "score": 0.5
      },
      {
        "name": "review",
        "alias": ["pr", "git flow", "代码审查"],
        "score": 0.5
      },
      {
        "name": "gayhub",
        "alias": ["github.com", "gitee.com"],
        "score": 1.5
      },
      {
        "name": "blog",
        "alias": [
          "个人博客",
          "博客地址",
          "csdn.net",
          "cnblogs.com",
          "juejin.im",
          "zhihu.com",
          "jianshu.com",
          "segmentfault.com",
          "oschina.net",
          "imooc.com",
          "51cto.com",
          "gitbooks.io"
        ],
        "score": 1.5
      }
    ]
  },
  {
    "name": "其他",
    "children": [
      {
        "name": "leader",
        "alias": ["组长", "负责人"],
        "score": 1
      },
      {
        "name": "mock",
        "score": 0.5
      },
      {
        "name": "git",
        "score": 0.5
      },
      {
        "name": "docker",
        "score": 0.5
      },
      {
        "name": "websocket",
        "alias": ["socket", "canvas", "charles"],
        "score": 0.5
      },
      {
        "name": "微服务",
        "score": 0.5
      },
      {
        "name": "加密",
        "alias": ["密文", "支付", "md5", "sha1", "签名", "白名单"],
        "score": 0.5
      },
      {
        "name": "重构",
        "alias": ["防抖", "去抖", "节流", "虚拟化", "懒加载"],
        "score": 0.5
      },
      {
        "name": "自定义",
        "alias": ["封装", "公共组件", "二次封装"],
        "score": 0.5
      },
      {
        "name": "python",
        "score": 0.5
      },
      {
        "name": "java",
        "score": 0.5
      },
      {
        "name": "go",
        "score": 0.5
      },
      {
        "name": "rust",
        "score": 0.5
      },
      {
        "name": "cpp",
        "score": 0.5
      },
      {
        "name": "swift",
        "score": 0.5
      },
      {
        "name": "kotlin",
        "score": 0.5
      },
      {
        "name": "scala",
        "score": 0.5
      },
      {
        "name": "mysql",
        "score": 0.5
      },
      {
        "name": "mongodb",
        "score": 0.5
      },
      {
        "name": "electron",
        "score": 0.5
      },
      {
        "name": "devops",
        "alias": ["shell", "nginx", "linux", "cicd"],
        "score": 0.5
      },
      {
        "name": "vim",
        "score": 0.5
      },
      {
        "name": "ssr",
        "score": 0.5
      },
      {
        "name": "bi",
        "score": 0.5,
        "alias": ["可视化", "大屏", "echarts", "d3"]
      },
      {
        "name": "three.js",
        "score": 0.5
      },
      {
        "name": "postman",
        "score": 0.5,
        "alias": ["charles", "fiddler"]
      },
      {
        "name": "leetcode",
        "score": 0.5,
        "alias": ["力扣", "leet code"]
      },
      {
        "name": "deno",
        "score": 0.5
      },
      {
        "name": "wasm",
        "score": 0.5,
        "alias": ["web assembly"]
      },
      {
        "name": "iframe",
        "alias": ["flash", "dreamweaver", "div+css", "v-for", "v-if"],
        "score": -1.5
      },
      {
        "name": "ajax",
        "alias": ["ajax", "jsonp", "ie6", "ie7", "ie8"],
        "score": -1
      },
      {
        "name": "jquery",
        "alias": ["jq", "zepto"],
        "score": -0.5
      },
      {
        "name": "svn",
        "alias": ["requirejs", "gulp", "grunt", "svn"],
        "score": -1
      },
      {
        "name": "积极乐观",
        "alias": ["为人和蔼", "吃苦耐劳", "上进", "了解"],
        "score": -0.5
      },
      {
        "name": "熟悉",
        "score": -0.3
      }
    ]
  }
]
