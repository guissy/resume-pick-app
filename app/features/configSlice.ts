import { createSlice } from '@reduxjs/toolkit';
// eslint-disable-next-line import/no-cycle
import { AppThunk, RootState } from '../store';
import { ConfigFile } from './type';
import initFile, { initDefaultFile, saveFile } from './configUtil';

export const configSlice = createSlice({
  name: 'config',
  initialState: { configs: {} as ConfigFile },
  reducers: {
    buildConfig: (state, action) => {
      if (Object.keys(state.configs).length === 0) {
        state.configs = action.payload;
      }
    },
    updateConfig: (state, action) => {
      Object.assign(state.configs, action.payload);
    },
    saveConfig: (state) => {
      saveFile(state.configs);
    },
    resetConfig: (state, action) => {
      state.configs = action.payload;
      saveFile(state.configs);
    },
  },
});

export const {
  updateConfig,
  buildConfig,
  saveConfig,
  resetConfig,
} = configSlice.actions;

export const initConfigAsync = (): AppThunk => (dispatch) =>
  initFile()
    .then((data) => dispatch(buildConfig(data)))
    // eslint-disable-next-line no-console
    .catch((e) => console.error(e));

export const resetConfigAsync = (): AppThunk => (dispatch) =>
  initDefaultFile()
    .then((data) => dispatch(resetConfig(data)))
    // eslint-disable-next-line no-console
    .catch((e) => console.error(e));

export default configSlice.reducer;

export const selectConfig = (state: RootState) => state.config.configs.default;
