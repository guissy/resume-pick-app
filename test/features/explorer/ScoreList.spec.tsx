import './mockGithubView';
import React from 'react';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { configureStore } from '@reduxjs/toolkit';
import * as configSlice from '../../../app/features/configSlice';
import * as scoreSlice from '../../../app/features/scoreSlice';
import ScoreList from '../../../app/features/explorer/ScoreList';
import * as exportExcelModule from '../../../app/features/parser/exportExcel';
import configDefault from '../../../app/constants/configDefault.json';

Enzyme.configure({ adapter: new Adapter() });
jest.useFakeTimers();

function setup(
  preloadedState = {
    config: {
      configs: {
        default: configDefault,
      },
    },
    score: {
      nameScore: [
        {
          name: '1',
          path: '1.pdf',
          workAge: 1,
          level: 'p4',
          levelValue: 0,
          school: '测大',
          score: 0,
          links: [],
          keywords: [
            {
              name: '',
              children: [
                {
                  name: 'vue',
                  score: 1,
                  gained: 1,
                  works: [
                    {
                      startDate: '2020-01-01',
                      endDate: '2020-11-01',
                      workContent: 'Vue',
                    },
                  ],
                },
              ],
            },
          ],
        },
      ],
    },
  }
) {
  const store = configureStore({
    reducer: {
      config: configSlice.default,
      score: scoreSlice.default,
    },
    preloadedState,
  });

  const getWrapper = () =>
    mount(
      <Provider store={store}>
        <Router>
          <ScoreList />
        </Router>
      </Provider>
    );
  const component = getWrapper();
  return {
    store,
    component,
    exportBtn: component.find('footer > button'),
    trashBtn: component.find('button.trash'),
  };
}

describe('ScoreList component', () => {
  it('should match exact snapshot', () => {
    const { store } = setup();
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <ScoreList />
          </Router>
        </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should second button should call exportExcel', () => {
    const { exportBtn } = setup();
    const exportExcelSpy = jest.spyOn(exportExcelModule, 'default');
    exportBtn.at(0).simulate('click');
    expect(exportExcelSpy).toBeCalled();
    exportExcelSpy.mockRestore();
  });

  it('should clearNameScore', () => {
    const { trashBtn } = setup();
    const clearNameScoreSpy = jest.spyOn(scoreSlice, 'clearNameScore');
    trashBtn.at(0).simulate('click');
    expect(clearNameScoreSpy).toBeCalled();
  });
});
