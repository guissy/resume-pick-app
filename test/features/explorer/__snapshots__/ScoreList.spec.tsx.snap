// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScoreList component should match exact snapshot 1`] = `
<div
  className="tableWrap"
  role="presentation"
>
  <dialog
    open={false}
    style={
      Object {
        "padding": 0,
        "position": "fixed",
        "top": 10,
        "width": "80vw",
        "zIndex": 100,
      }
    }
  >
    <div
      className="wrap"
    >
      <header
        style={
          Object {
            "textAlign": "right",
          }
        }
      >
        <label
          className="searchWrap"
          htmlFor="search"
        >
          <i
            className="fa fa-search"
          />
          <span
            className="count"
            style={
              Object {
                "visibility": "hidden",
              }
            }
          >
            0
          </span>
          <input
            className="search"
            id="search"
            onChange={[Function]}
            placeholder="输入关键字搜索简历"
            type="search"
            value=""
          />
        </label>
        <button
          onClick={[Function]}
          style={
            Object {
              "background": "none",
              "border": "1px solid #fff",
              "color": "#fff",
              "cursor": "pointer",
              "fontSize": 14,
            }
          }
          type="button"
        >
          使用本地软件打开此文件
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          <i
            className="fa fa-times fa-2x"
          />
        </button>
      </header>
      <article
        className="article"
      >
        <p
          style={
            Object {
              "marginBlockEnd": 5,
              "marginBlockStart": 5,
            }
          }
        >
          <span />
        </p>
      </article>
    </div>
  </dialog>
  <header
    className="header"
  >
    <label
      htmlFor="showFull"
    >
      <input
        checked={true}
        id="showFull"
        onChange={[Function]}
        type="checkbox"
      />
      Show Full
    </label>
    <label
      htmlFor="gitRepo"
    >
      <input
        checked={false}
        id="gitRepo"
        onChange={[Function]}
        type="checkbox"
      />
      Git Repo
    </label>
    <button
      className="trash"
      onClick={[Function]}
      type="button"
    >
      <span>
        清空
      </span>
      <i
        className="fa fa-trash"
      />
    </button>
    <label
      className="searchWrap"
      htmlFor="search"
    >
      <i
        className="fa fa-search"
      />
      <input
        className="search"
        id="search"
        onChange={[Function]}
        placeholder="输入关键字搜索简历"
        type="search"
        value=""
      />
    </label>
    <div
      className="opts"
    >
      <label
        htmlFor="optCheckbox"
      >
        <input
          checked={true}
          id="optCheckbox"
          onChange={[Function]}
          type="checkbox"
        />
        全选
      </label>
      <ul
        className="icons"
      >
        <li
          style={
            Object {
              "backgroundColor": "#333",
            }
          }
        >
          <button
            className="opt"
            onClick={[Function]}
            style={
              Object {
                "filter": "unset",
              }
            }
            type="button"
          >
            <img
              alt="RN"
              src="test-file-stub"
            />
          </button>
        </li>
        <li
          style={
            Object {
              "backgroundColor": "#333",
            }
          }
        >
          <button
            className="opt"
            onClick={[Function]}
            style={
              Object {
                "filter": "unset",
              }
            }
            type="button"
          >
            <img
              alt="flutter"
              src="test-file-stub"
            />
          </button>
        </li>
        <li
          style={
            Object {
              "backgroundColor": "#333",
            }
          }
        >
          <button
            className="opt"
            onClick={[Function]}
            style={
              Object {
                "filter": "unset",
              }
            }
            type="button"
          >
            <img
              alt="react"
              src="test-file-stub"
            />
          </button>
        </li>
        <li
          style={
            Object {
              "backgroundColor": "#333",
            }
          }
        >
          <button
            className="opt"
            onClick={[Function]}
            style={
              Object {
                "filter": "unset",
              }
            }
            type="button"
          >
            <img
              alt="vue"
              src="test-file-stub"
            />
          </button>
        </li>
        <li
          style={
            Object {
              "backgroundColor": "#333",
            }
          }
        >
          <button
            className="opt"
            onClick={[Function]}
            style={
              Object {
                "filter": "unset",
              }
            }
            type="button"
          >
            <img
              alt="angular"
              src="test-file-stub"
            />
          </button>
        </li>
        <li
          style={
            Object {
              "backgroundColor": "#333",
            }
          }
        >
          <button
            className="opt"
            onClick={[Function]}
            style={
              Object {
                "filter": "unset",
              }
            }
            type="button"
          >
            <img
              alt="wxapp"
              src="test-file-stub"
            />
          </button>
        </li>
        <li
          style={
            Object {
              "backgroundColor": "#333",
            }
          }
        >
          <button
            className="opt"
            onClick={[Function]}
            style={
              Object {
                "filter": "unset",
              }
            }
            type="button"
          >
            <img
              alt="node"
              src="test-file-stub"
            />
          </button>
        </li>
        <li
          style={
            Object {
              "backgroundColor": "#333",
            }
          }
        >
          <button
            className="opt"
            onClick={[Function]}
            style={
              Object {
                "filter": "unset",
              }
            }
            type="button"
          >
            <img
              alt="es6"
              src="test-file-stub"
            />
          </button>
        </li>
        <li
          style={
            Object {
              "backgroundColor": "#333",
            }
          }
        >
          <button
            className="opt"
            onClick={[Function]}
            style={
              Object {
                "filter": "unset",
              }
            }
            type="button"
          >
            <img
              alt="css3"
              src="test-file-stub"
            />
          </button>
        </li>
        <li
          style={
            Object {
              "backgroundColor": "#333",
            }
          }
        >
          <button
            className="opt"
            onClick={[Function]}
            style={
              Object {
                "filter": "unset",
              }
            }
            type="button"
          >
            <img
              alt="typescript"
              src="test-file-stub"
            />
          </button>
        </li>
        <li
          style={
            Object {
              "backgroundColor": "#333",
            }
          }
        >
          <button
            className="opt"
            onClick={[Function]}
            style={
              Object {
                "filter": "unset",
              }
            }
            type="button"
          >
            <img
              alt="graphql"
              src="test-file-stub"
            />
          </button>
        </li>
        <li
          style={
            Object {
              "backgroundColor": "#333",
            }
          }
        >
          <button
            className="opt"
            onClick={[Function]}
            style={
              Object {
                "filter": "unset",
              }
            }
            type="button"
          >
            <img
              alt="test"
              src="test-file-stub"
            />
          </button>
        </li>
        <li
          style={
            Object {
              "backgroundColor": "#333",
            }
          }
        >
          <button
            className="opt"
            onClick={[Function]}
            style={
              Object {
                "filter": "unset",
              }
            }
            type="button"
          >
            <img
              alt="gayhub"
              src="test-file-stub"
            />
          </button>
        </li>
        <li
          style={
            Object {
              "backgroundColor": "#333",
            }
          }
        >
          <button
            className="opt"
            onClick={[Function]}
            style={
              Object {
                "filter": "unset",
              }
            }
            type="button"
          >
            <img
              alt="lint"
              src="test-file-stub"
            />
          </button>
        </li>
        <li
          style={
            Object {
              "backgroundColor": "#333",
            }
          }
        >
          <button
            className="opt"
            onClick={[Function]}
            style={
              Object {
                "filter": "unset",
              }
            }
            type="button"
          >
            <img
              alt="electron"
              src="test-file-stub"
            />
          </button>
        </li>
        <li
          style={
            Object {
              "backgroundColor": "#333",
            }
          }
        >
          <button
            className="opt"
            onClick={[Function]}
            style={
              Object {
                "filter": "unset",
              }
            }
            type="button"
          >
            <img
              alt="git"
              src="test-file-stub"
            />
          </button>
        </li>
        <li
          style={
            Object {
              "backgroundColor": "#333",
            }
          }
        >
          <button
            className="opt"
            onClick={[Function]}
            style={
              Object {
                "filter": "unset",
              }
            }
            type="button"
          >
            <img
              alt="docker"
              src="test-file-stub"
            />
          </button>
        </li>
        <li
          style={
            Object {
              "backgroundColor": "#333",
            }
          }
        >
          <button
            className="opt"
            onClick={[Function]}
            style={
              Object {
                "filter": "unset",
              }
            }
            type="button"
          >
            <img
              alt="python"
              src="test-file-stub"
            />
          </button>
        </li>
        <li
          style={
            Object {
              "backgroundColor": "#333",
            }
          }
        >
          <button
            className="opt"
            onClick={[Function]}
            style={
              Object {
                "filter": "unset",
              }
            }
            type="button"
          >
            <img
              alt="java"
              src="test-file-stub"
            />
          </button>
        </li>
        <li
          style={
            Object {
              "backgroundColor": "#333",
            }
          }
        >
          <button
            className="opt"
            onClick={[Function]}
            style={
              Object {
                "filter": "unset",
              }
            }
            type="button"
          >
            <img
              alt="go"
              src="test-file-stub"
            />
          </button>
        </li>
        <li
          style={
            Object {
              "backgroundColor": "#333",
            }
          }
        >
          <button
            className="opt"
            onClick={[Function]}
            style={
              Object {
                "filter": "unset",
              }
            }
            type="button"
          >
            <img
              alt="kotlin"
              src="test-file-stub"
            />
          </button>
        </li>
        <li
          style={
            Object {
              "backgroundColor": "#333",
            }
          }
        >
          <button
            className="opt"
            onClick={[Function]}
            style={
              Object {
                "filter": "unset",
              }
            }
            type="button"
          >
            <img
              alt="mongodb"
              src="test-file-stub"
            />
          </button>
        </li>
      </ul>
    </div>
  </header>
  <table
    className="table"
  >
    <thead>
      <tr>
        <td
          className="td"
          style={
            Object {
              "width": 32,
            }
          }
        >
          序号
        </td>
        <td
          className="td"
          style={
            Object {
              "width": 260,
            }
          }
        >
          文件
        </td>
        <td
          className="td sort"
          style={
            Object {
              "width": 50,
            }
          }
        >
          <button
            className="sortBtn"
            onClick={[Function]}
            type="button"
          >
            <span>
              经验
            </span>
            <div
              className="caretInit"
            >
              <i
                className="fa fa-caret-up "
              />
              <i
                className="fa fa-caret-down "
              />
            </div>
          </button>
        </td>
        <td
          className="td sort"
        >
          <button
            className="sortBtn"
            onClick={[Function]}
            type="button"
          >
            <span>
              分数
            </span>
            <div
              className="caretInit"
            >
              <i
                className="fa fa-caret-up "
              />
              <i
                className="fa fa-caret-down "
              />
            </div>
          </button>
        </td>
        <td
          className="td"
          colSpan={1}
          style={
            Object {
              "width": 150,
            }
          }
        >
          github
        </td>
        <td
          className="td"
          colSpan={1}
          style={
            Object {
              "width": "auto",
            }
          }
        >
          关键字
        </td>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          className="td"
        >
          1
          <br />
          <button
            className="trash"
            onClick={[Function]}
            type="button"
          >
            <i
              className="fa fa-trash"
            />
          </button>
        </td>
        <td
          className="td"
        >
          <button
            className="link"
            onClick={[Function]}
            type="button"
          >
            <span
              className="namePhone"
            >
              1
            </span>
            <span
              className="namePhone"
            />
            <span
              className="namePhone"
            >
              测大
            </span>
            <span
              className="namePhone"
            />
            <ul
              className="icons"
            >
              <li
                style={
                  Object {
                    "backgroundColor": "rgba(235, 235, 235, 1.2)",
                  }
                }
              >
                <img
                  alt="vue"
                  src="test-file-stub"
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                />
              </li>
            </ul>
          </button>
        </td>
        <td
          className="td"
        >
          1年
        </td>
        <td
          className="score"
        >
          0.00
        </td>
        <td
          className="td"
        >
          <span
            className="level level4"
            title="0"
          >
            p4
          </span>
        </td>
        <td
          className="td"
        >
          <div>
            <svg
              height={120}
              width={360}
            />
          </div>
        </td>
      </tr>
    </tbody>
  </table>
  <footer
    className="footer"
  >
    <button
      className="exportExcel"
      onClick={[Function]}
      type="button"
    >
      导出 Excel
    </button>
  </footer>
</div>
`;
